### PR TITLE
Make resource pools static to the files they are in.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: src/resources/ethernet_esp32.cc [hunk 1]
```diff
@@ -56,7 +56,7 @@ enum {
 const int kInvalidEthernet = -1;
 
 // Only allow one instance of WiFi running.
-ResourcePool<int, kInvalidEthernet> ethernet_pool(
+static ResourcePool<int, kInvalidEthernet> ethernet_pool(
   0
```
Comment (kasperl) [inline src/resources/ethernet_esp32.cc:60]: 2 or 4 space indent for this? Looks like we're not very consistent across the code base. I think we mostly use 4.

File: src/resources/i2c_esp32.cc [hunk 2]
```diff
@@ -39,7 +39,7 @@ const i2c_port_t kInvalidPort = i2c_port_t(-1);
 #error "SOC_I2C_NUM not defined"
 #endif
 
-ResourcePool<i2c_port_t, kInvalidPort> i2c_ports(
+static ResourcePool<i2c_port_t, kInvalidPort> i2c_ports(
    I2C_NUM_0
```
Comment (kasperl) [inline src/resources/i2c_esp32.cc:43]: 3 space indent is a new one.

File: src/resources/spi_esp32.cc [hunk 3]
```diff
@@ -32,11 +32,11 @@
...
 
-ResourcePool<int, 0> dma_channels(1, 2);
+static ResourcePool<int, 0> dma_channels(1, 2);
 
 const spi_host_device_t kInvalidHostDevice = spi_host_device_t(-1);
 
-ResourcePool<spi_host_device_t, kInvalidHostDevice> spi_host_devices(
+static ResourcePool<spi_host_device_t, kInvalidHostDevice> spi_host_devices(
 #ifdef CONFIG_IDF_TARGET_ESP32S3
   SPI2_HOST,
```
Comment (kasperl) [inline src/resources/spi_esp32.cc:41]: Here it is 2.

File: src/resources/uart_esp32.cc [hunk 4]
```diff
@@ -52,7 +52,7 @@ const int kReadState = 1 << 0;
 const int kErrorState = 1 << 1;
 const int kWriteState = 1 << 2;
 
-ResourcePool<uart_port_t, kInvalidUartPort> uart_ports(
+static ResourcePool<uart_port_t, kInvalidUartPort> uart_ports(
   // Uart 0 is reserved serial communication (stdout).
 #if SOC_UART_NUM > 2
   UART_NUM_2,
```
Comment (kasperl) [inline src/resources/uart_esp32.cc:58]: Here it is 2.

**Review by kasperl — APPROVED** [review]
Looks good to me.

**Comment by floitsch** [comment]
I updated the Indentation.

**Pull request merged**